pkg/kube/helpers.go

```diff
@@ -193,15 +193,19 @@ func CreateOrUpdateAnnotated(ctx context.Context, c client.Client, obj runtime.O
 	return true, nil
 }
 
-func mutate(f MutateFn, key client.ObjectKey, obj runtime.Object) error {
+func mutate(f MutateFn, key client.ObjectKey, obj client.Object) error {
 	if err := f(); err != nil {
 		return err
 	}
 
-	if newKey, err := client.ObjectKeyFromObject(obj); err != nil || key != newKey {
-		return fmt.Errorf("MutateFn cannot mutate object name and/or object namespace")
-	}
+	// TODO uncertain how to do this
```

Comment:
There should be a helper floating around somewhere? It may be sufficient to just save the name & namespace and then compare them after calling `f`?